mesonbuild/ast/interpreter.py

```diff
@@ -60,39 +75,136 @@
         UMinusNode,
     )
 
-class DontCareObject(MesonInterpreterObject):
-    pass
-
-class MockExecutable(MesonInterpreterObject):
-    pass
+# `IntrospectionFile` is to the `IntrospectionInterpreter` what `File` is to the normal `Interpreter`.
+@dataclass
+class IntrospectionFile(HoldableObject):
+    subdir: str
+    rel: str
 
-class MockStaticLibrary(MesonInterpreterObject):
-    pass
+    def to_abs_path(self, root_dir: Path) -> Path:
+        return (root_dir / self.subdir / self.rel).resolve()
 
-class MockSharedLibrary(MesonInterpreterObject):
-    pass
+    def __hash__(self) -> int:
+        return hash((self.__class__.__name__, self.subdir, self.rel))
```

Comment:
Why do you need the class name in the hash function?

Reply:
Because Im afraid that otherwise, sometime in the future, a hash collision like the one below might occur:


```
class IntrospectionFile:

    def __hash__(self) -> int:
        return hash((self.subdir, self.rel))


class Foo:

    def __hash__(self) -> int:
        return hash((self.subdir, self.rel))

file = IntrospectionFile()
file.subdir = "abc"
file.rel = "def"


foo = Foo()
foo.subdir = "abc"
foo.rel = "def"


print(hash(file))
print(hash(foo))
```